src/mongocxx/options/count.cpp

```diff
@@ -33,6 +36,8 @@ count& count::limit(std::int64_t limit) {
 }
 
 count& count::max_time(std::chrono::milliseconds max_time) {
+    if (!is_int32_duration<std::chrono::milliseconds>(max_time))
```

Comment:
I'm pretty OK with the plan of validating at option construction time. I have one *small* nit. Which is that really, the problem isn't when you set the options, it is when you try to apply them against an API that can't support them, being the C driver.

With things this way, that relationship is masked: we are enforcing a constraint *here*, that is really an implementation detail relevant over *there*.

I'm not sure how much it matters really though. Just pointing it out in case you have any thoughts.